airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py

```diff
@@ -76,9 +78,33 @@ def records_with_state(records, state, stream_mapping, state_cursor_paths) -> It
         yield record_value, state_value, stream_name
 
 
+def compare_cursor_with_threshold(record_value, state_value, threshold_days) -> bool:
+    """
+    Checks if the record's cursor value is older or equal to the state cursor value.
+
+    If the threshold_days option is set, the values will be converted to dates so that the time-based offset can be applied.
+    :raises dateutil.parser._parser.ParserError: if threshold_days is passed with non-date cursor values.
+    """
+    if threshold_days:
+        record_date_value = record_value if isinstance(record_value, datetime) else dateutil.parser.parse(record_value)
+        state_date_value = state_value if isinstance(state_value, datetime) else dateutil.parser.parse(state_value)
+
+        return record_date_value >= (state_date_value - timedelta(days=threshold_days))
+
+    return record_value >= state_value
+
+
 @pytest.mark.default_timeout(20 * 60)
 class TestIncremental(BaseTest):
-    def test_two_sequential_reads(self, connector_config, configured_catalog_for_incremental, cursor_paths, docker_runner: ConnectorRunner):
+    def test_two_sequential_reads(
+        self,
+        inputs: IncrementalConfig,
+        connector_config: SecretDict,
+        configured_catalog_for_incremental: ConfiguredAirbyteCatalog,
+        cursor_paths: dict[str, list[str]],
+        docker_runner: ConnectorRunner,
+    ):
+        threshold_days = getattr(inputs, "threshold_days") or 0
```

Comment:
Looks like the default is used if the attribute doesn't exist, rather than when the value itself is `None`

https://docs.python.org/3/library/functions.html#getattr